code/hud/hudlock.cpp

```diff
@@ -807,36 +776,16 @@ void hud_lock_acquire_uncaged_subsystem(weapon_info *wip, lock_info *lock, float
 	Assert( lock->obj->type == OBJ_SHIP );
 
 	ship *sp = &Ships[lock->obj->instance];
-	ship_subsys	*ss;
 
 	float ss_dot;
-	vec3d ss_pos;
-	vec3d vec_to_target;
 
 	int current_num_locks = 0;
 
 	if ( Ship_info[sp->ship_info_index].is_big_or_huge() ) {
-		for (ss = GET_FIRST(&sp->subsys_list); ss != END_OF_LIST(&sp->subsys_list); ss = GET_NEXT(ss) ) {
-			if (ss->flags[Ship::Subsystem_Flags::Untargetable]) {
-				continue;
-			}
-
-			get_subsystem_world_pos(lock->obj, ss, &ss_pos);
-
-			if ( !hud_lock_world_pos_in_range(&ss_pos, &vec_to_target)  ) {
-				continue;
-			}
-
-			vm_vec_normalize(&vec_to_target);
-			ss_dot = vm_vec_dot(&Player_obj->orient.vec.fvec, &vec_to_target);
+		for (ship_subsys* ss = GET_FIRST(&sp->subsys_list); ss != END_OF_LIST(&sp->subsys_list); ss = GET_NEXT(ss) ) {
 
-			if ( ss_dot < wip->lock_fov ) {
+			if (!weapon_multilock_can_lock_on_subsys(Player_obj, lock->obj, ss, wip, &ss_dot))
 				continue;
-			}
-
-			if ( !ship_subsystem_in_sight(lock->obj, ss, &Eye_position, &ss_pos) ) {
```

Comment:
I wonder how much of an impact changing from Eye_position to the object's position will have on the targeting code.  In most cases it shouldn't matter, but there *are* ships where the eye position is noticeably offset.  Can you change the function to accept the eye position?

Reply:
Ah, I hadn't even noticed that change, yeah that should be possible.